console/console-init/ui/src/pages/AddressSpaceDetail/AddressList/AddressListFilter.tsx

```diff
@@ -156,11 +167,21 @@ export const AddressListFilter: React.FunctionComponent<IAddressListFilterProps>
       response &&
       response.data &&
       response.data.addresses &&
-      response.data.addresses.addresses
+      response.data.addresses.addresses &&
+      response.data.addresses.addresses.length > 0
     ) {
       //To display dropdown if fetched records are less than 100 in count.
-      if (response.data.addresses.total > 100) {
-        setNameOptions([]);
+      if (
+        response.data.addresses.total >
+        MAX_ITEM_TO_DISPLAY_IN_TYPEAHEAD_DROPDOWN
+      ) {
+        let list = [];
+        for (let i = 0; i < 10; i++) {
```

Comment:
I think it's better to store 10 in a const variable in a global file and import it from there, as it's being used in multiple files.

Reply:
Also, this logic here is repetitive and we can have a common function which just accepts a list of records and push the 10 records in an array